src/pages/settings/Profile/ProfilePage.js

```diff
@@ -152,55 +133,85 @@ class ProfilePage extends Component {
      * @param {Object} avatar
      */
     updateAvatar(avatar) {
-        this.setState({avatar, isAvatarChanged: true});
-    }
+        this.avatar = _.isUndefined(avatar) ? {uri: ReportUtils.getDefaultAvatar(this.props.currentUserPersonalDetails.login)} : avatar;
 
-    /**
-     * Replaces the user's current avatar image with a default avatar.
-     */
-    deleteAvatar() {
-        PersonalDetails.deleteAvatar();
-        this.setState({avatar: {uri: ReportUtils.getDefaultAvatar(lodashGet(this.props.currentUserPersonalDetails, 'login'))}});
+        this.setState({isAvatarChanged: true});
     }
 
     /**
      * Submit form to update personal details
+     * @param {Object} values
+     * @param {String} values.firstName
+     * @param {String} values.lastName
+     * @param {String} values.pronouns
+     * @param {Boolean} values.isAutomaticTimezone
+     * @param {String} values.timezone
+     * @param {String} values.selfSelectedPronoun
      */
-    updatePersonalDetails() {
-        if (!this.validateInputs()) {
-            return;
-        }
-
+    updatePersonalDetails(values) {
         // Check if the user has modified their avatar
-        if ((lodashGet(this.props.currentUserPersonalDetails, 'avatar') !== this.state.avatar.uri) && this.state.isAvatarChanged) {
-            PersonalDetails.setAvatar(this.state.avatar);
+        if ((this.props.currentUserPersonalDetails.avatar !== this.avatar.uri) && this.state.isAvatarChanged) {
+            // If the user removed their profile photo, replace it accordingly with the default avatar
+            if (this.avatar.uri.includes('/images/avatars/avatar')) {
+                PersonalDetails.deleteAvatar(this.avatar.uri);
+            } else {
+                PersonalDetails.setAvatar(this.avatar);
+            }
 
             // Reset the changed state
             this.setState({isAvatarChanged: false});
         }
 
         PersonalDetails.updateProfile(
-            this.state.firstName.trim(),
-            this.state.lastName.trim(),
-            this.state.pronouns.trim(),
+            values.firstName.trim(),
+            values.lastName.trim(),
+            (this.state.hasSelfSelectedPronouns) ? values.selfSelectedPronoun.trim() : values.pronouns.trim(),
             {
-                automatic: this.state.isAutomaticTimezone,
-                selected: this.state.selectedTimezone,
+                automatic: values.isAutomaticTimezone,
+                selected: values.timezone,
```

Comment:
>  - automatic: this.state.isAutomaticTimezone,
> - selected: this.state.selectedTimezone,
> + automatic: values.isAutomaticTimezone,
> + selected: values.timezone,

Sorry - what needs to be done here? Isn't it already using `values` instead of `this.state`?